### PR TITLE
Recent Scrutinizer Messages

### PR DESCRIPTION
**Pull request by oleibman**
Get rid of them where possible.

This is:

- [ ] a bugfix
- [ ] a new feature
- [ ] refactoring
- [ ] additional unit tests

Checklist:

- [ ] Changes are covered by unit tests
  - [x] Changes are covered by existing unit tests
  - [ ] New unit tests have been added
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [ ] CHANGELOG.md contains a short summary of the change and a link to the pull request if applicable
- [ ] Documentation is updated as necessary

### Why this change is needed?

Provide an explanation of why this change is needed, with links to any Issues (if appropriate).
If this is a bugfix or a new feature, and there are no existing Issues, then please also create an issue that will make it easier to track progress with this PR.

### CODE REVIEW

File: src/PhpSpreadsheet/Worksheet/Worksheet.php [hunk 1]
```diff
@@ -3442,7 +3442,7 @@ public function getHashCode()
      * @param string $range Range to extract title from
      * @param bool $returnRange Return range? (see example)
      *
-     * @return ($range is non-empty-string ? ($returnRange is true ? array{0: string, 1: string} : string) : ($returnRange is true ? array{0: null, 1: null} : null))
+     * @return ($range is non-empty-string ? ($returnRange is true ? array{0: string, 1: string} : string) : ($returnRange is true ? array{0: null, 1: null} : null)) Scrutinizer does not understand this but it helps Phpstan
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Worksheet/Worksheet.php:3445]:
```suggestion
     * @return ($range is non-empty-string ? ($returnRange is true ? array{0: string, 1: string} : string) : ($returnRange is true ? array{0: null, 1: null} : null))
```

Don't think we should have those kind of comment that end up in end-user documentation.

**Review by PowerKiKi — COMMENTED** [review]
On a more general note, I feel nowaday PHPStan is _way_ better than Scrutinizer static analyzer and we probably should not spend too much time on accommodating Scrutinizer false positives.

While Scrutinizer is still able to catch things that PHPStan cannot, so it makes sense to keep it, I don't think we should riddle our code with `@scrutinizer ignore-type` or similar. Instead we can ignore whole range of errors directly from Scrutinizer UI. Typically `$drawing->setImageResource(/** @scrutinizer ignore-type */ $gdImage);` is a false positive that clutter our code for nothing.

**Comment by PowerKiKi** [comment]
I gave you access to Scrutinizer:

image

**Comment by oleibman** [comment]
@PowerKiKi converting to draft so that you can carry on with your work in #3743. As for the comments in the doc-block, I couldn't figure out how else to indicate that Scrutinizer cares but we consciously don't care (as opposed to we've never even looked at the message).

I agree that Phpstan is way better than Scrutinizer in its analysis.

**Comment by PowerKiKi** [comment]
> couldn't figure out how else to indicate that Scrutinizer cares but we consciously don't care

Now you should be able to do that on https://scrutinizer-ci.com/g/PHPOffice/PhpSpreadsheet/issues/master

**Comment by oleibman** [comment]
Superseded by #3743.

**Pull request closed without merging**